capstone/src/main/java/com/google/sps/servlets/GroupPostDataServlet.java

```diff
@@ -36,18 +36,17 @@
 import com.google.sps.Objects.Post;
 import com.google.sps.Objects.Comment;
 import com.google.sps.Objects.response.PostResponse;
+import error.ErrorHandler;
 
 @WebServlet("/group-post")
 public class GroupPostDataServlet extends HttpServlet {
 
+  private ErrorHandler errorHandler = new ErrorHandler();
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
-    String userId = "";
-    UserService userService = UserServiceFactory.getUserService();
-    if (userService.isUserLoggedIn()) {
-      userId = userService.getCurrentUser().getUserId();
-    }
+    String userId = this.getUserId(response);
```

Comment:
Same comment from other file: Lucy made an AuthenticatedServlet that you can extend instead of the HttpServlet that will do the user Authentication for you: https://github.com/googleinterns/step100-2020/blob/master/capstone/src/main/java/com/google/sps/servlets/AuthenticatedServlet.java
So you just need to make sure to extend the AuthenticatedServlet and you can remove an user login related code. Also note the doPost/doGet change to give you access to the userID